src/server.cpp

```diff
@@ -1125,25 +1124,29 @@ void CServer::MixEncodeTransmitData ( const int iChanCnt,
                     // mono
                     for ( i = 0; i < iServerFrameSizeSamples; i++ )
                     {
-                        vecdIntermProcBuf[i] += vecsData[i] * dGain;
+                        vecfIntermProcBuf[i] += vecfData[i] * dGain;
                     }
                 }
                 else
                 {
                     // stereo: apply stereo-to-mono attenuation
                     for ( i = 0, k = 0; i < iServerFrameSizeSamples; i++, k += 2 )
                     {
-                        vecdIntermProcBuf[i] += dGain *
-                            ( static_cast<double> ( vecsData[k] ) + vecsData[k + 1] ) / 2;
+                        vecfIntermProcBuf[i] += dGain *
+                            ( static_cast<float> ( vecfData[k] ) + vecfData[k + 1] ) / 2;
                     }
                 }
             }
         }
 
-        // convert from double to short with clipping
+        // When adding multiple sound sources together
+        // the resulting signal level may exceed the maximum
+        // audio range which is from -1.0f to 1.0f inclusivly.
+        // Clip the intermediate sound buffer to be within
+        // the expected range
         for ( i = 0; i < iServerFrameSizeSamples; i++ )
         {
-            vecsSendData[i] = Double2Short ( vecdIntermProcBuf[i] );
+            vecfSendData[i] = clipFloat ( vecfIntermProcBuf[i] );
```

Comment:
Is it more efficient to pass a reference to the vector to the clipper and clip the entire vector in the method called, rather than a call per vector entry?